### DISCUSSION TITLE
UI load tests + integration with Playwright

### DISCUSSION THREAD

**Discussion by Romarionijim** [locustio/locust#2799] [Q&A]
Hello, I'm an automation engineer, and I am a but new to load testing, I'm researching automation load testing tools to implement in my current workplace, and I have a few small questions regarding Locust.

I saw in the docs a small test example, but wanted to understand more, in order to use Locust for load testing, can I utilize an already existing framework written in playwright? meaning if I have existing UI page object model framework written in Python and playwright, can I reuse them for load tests?

does locust support load test for UI? because in the example it seems it support API endpoints and I think I'm missing the UI load testing to simulate thousands of users doing the same action in the UI.

Thank you in advance!

**Comment by cyberw** [comment]
Hi! Locust itself doesnt support UI testing, but because Locust is "just python", you can interact with Playwright from it.

There's an implementation (maybe not the best or most performant) in locust-plugins: https://github.com/SvenskaSpel/locust-plugins/blob/master/examples/playwright_ex.py (its not really supported though, and you are on your own if you run into problems)

**Reply by Romarionijim** [reply]
@cyberw locust focuses more on backend load testing instead of ui load testing correct? I was wondering if I could leverage an existing framework written in playwright and creating load tests from them.

**Reply by cyberw** [reply]
Locust is mainly used for testing at ”request level”, but can technically test whatever python can test :)

You could use an existing playwright framework, but you’d need to integrate it with locust, maybe based on the PlaywrightUser in locust plugins.

**Reply by Romarionijim** [reply]
And this is maintained by you guys? https://github.com/SvenskaSpel/locust-plugins/blob/master/examples/playwright_ex.py

**Reply by cyberw** [reply]
Yes and no. I wrote it, but it’s not really maintained. Might work for you, but you may have issues and it’s not at the top of my list of things to fix :) But definitely try it out!
